### PR TITLE
fix(BUX-310): trim sync.Results to last 20 results

### PR DESCRIPTION
**Pull request by arkadiuszos4chain**
# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/BuxOrg/bux/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/BuxOrg/bux/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [x] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: model_sync_transactions_test.go [hunk 1]
```diff
@@ -101,3 +101,27 @@ func Test_areParentsBroadcast(t *testing.T) {
 		})
 	}
 }
+
+func TestSyncTransaction_SaveHook(t *testing.T) {
+	t.Parallel()
+
+	t.Run("trim Results to last 20 messages", func(t *testing.T) {
+		ctx, client, deferMe := CreateTestSQLiteClient(t, false, true, WithCustomTaskManager(&taskManagerMockBase{}))
```
Comment (kuba-4chain) [inline model_sync_transactions_test.go:109]: Our Code Standards say that we should have
// Given
...
// When
...
// Then
...
in our tests :)

File: model_sync_transactions.go [hunk 2]
```diff
@@ -122,6 +122,21 @@ func (m *SyncTransaction) AfterCreated(ctx context.Context) error {
...
 
+func (m *SyncTransaction) BeforeUpdating(ctx context.Context) error {
+	m.DebugLog("starting: " + m.Name() + " BeforeUpdate hook...")
+
+	// Trim the results to the last 20
+	maxResultsLength := 20
+
+	ln := len(m.Results.Results)
+	if ln > maxResultsLength {
+		m.Results.Results = m.Results.Results[ln-maxResultsLength:]
```
Comment (dorzepowski) [inline model_sync_transactions.go:133]: Maybe add log message about trimming the results.

Comment (arkadiuszos4chain) [reply model_sync_transactions.go:133]: Warning, info?

Comment (arkadiuszos4chain) [reply model_sync_transactions.go:137]: Warning added

**Review by pawellewandowski98 — APPROVED** [review]

**Review by dorzepowski — APPROVED** [review]

**Pull request merged**